ResearchKit/Common/ORKStepNavigationRule.m

```diff
@@ -450,3 +451,110 @@ - (NSUInteger)hash {
 }
 
 @end
+
+
+@implementation ORKStepModifier
+
+- (void)modifyStep:(ORKStep *)step withTaskResult:(ORKTaskResult *)taskResult {
+    @throw [NSException exceptionWithName:NSGenericException reason:@"You should override this method in a subclass" userInfo:nil];
+}
+
+#pragma mark NSSecureCoding
+
++ (BOOL)supportsSecureCoding {
+    return YES;
+}
+
+- (instancetype)initWithCoder:(NSCoder *)aDecoder {
+    return [super init];
+}
+
+- (void)encodeWithCoder:(NSCoder *)aCoder {
+}
+
+#pragma mark NSCopying
+
+- (instancetype)copyWithZone:(NSZone *)zone {
+    return [[[self class] allocWithZone:zone] init];
+}
+
+- (BOOL)isEqual:(id)object {
+    if ([self class] != [object class]) {
+        return NO;
+    }
+    return YES;
+}
+
+@end
+
+
+@implementation ORKKeyValueStepModifier
+
+- (instancetype)initWithResultPredicate:(NSPredicate *)resultPredicate
+                           keyValueMap:(NSDictionary<NSString *, NSObject *> *)keyValueMap {
+    ORKThrowInvalidArgumentExceptionIfNil(resultPredicate);
+    ORKThrowInvalidArgumentExceptionIfNil(keyValueMap);
+    if (self) {
+        _resultPredicate = resultPredicate;
+        _keyValueMap = keyValueMap;
+    }
+    return self;
+}
+
+- (void)modifyStep:(ORKStep *)step withTaskResult:(ORKTaskResult *)taskResult {
+    
+    // The predicate can either have:
+    // - an ORKResultPredicateTaskIdentifierVariableName variable which will be substituted by the ongoign task identifier;
```

Comment:
Sorry, another typo in `substituionVariables` (-> `substitutionVariables`), my fault again!